src/main/js/CIStatusBar.tsx

```diff
@@ -56,37 +57,51 @@ export default class CIStatusBar extends React.Component<Props, State> {
     this.props.repository._links.ciStatus && this.fetchCIStatus();
   }
 
-  fetchCIStatus = () => {
-    const { pullRequest } = this.props;
-    const url = (pullRequest._links.ciStatus as Link)?.href;
-    if (url) {
-      this.setState({
-        loading: true
+  fetchCIStatusFromUrl = (url?: string) => {
+    apiClient
+      .get(url)
+      .then(response => response.json())
+      .then(json => {
+        this.setState(
+          {
+            ciStatus: json._embedded.ciStatus,
+            loading: false
+          },
+          this.setStatus
+        );
+      })
+      .catch(error => {
+        if (error instanceof NotFoundError) {
+          this.setState({
+            loading: false
+          });
+        } else {
+          this.setState({
+            error,
```

Comment:
Use react-query instead?

Reply:
This would lead to a bigger refactoring of old code I liked to avoid.